packages/media-utils/src/utils/upload-media.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { createBlobURL, revokeBlobURL } from '@wordpress/blob';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	AdditionalData,
+	Attachment,
+	OnChangeHandler,
+	OnErrorHandler,
+	OnSuccessHandler,
+} from './types';
+import { uploadToServer } from './upload-to-server';
+import { validateMimeType } from './validate-mime-type';
+import { validateMimeTypeForUser } from './validate-mime-type-for-user';
+import { validateFileSize } from './validate-file-size';
+import { UploadError } from './upload-error';
+
+const noop = () => {};
```

Comment:
We could easily get away of using `noop`s by calling `onError` with optional chaining:

```
onError?.()
```